src/cup1d/data/data_Karacayli2023.py

```diff
@@ -0,0 +1,71 @@
+import os
+
+import pandas
+import numpy as np
+
+from cup1d.data.base_p1d_data import BaseDataP1D, _drop_zbins
+
+class P1D_Karacayli2023(BaseDataP1D):
+
+    def __init__(self,diag_cov=False,kmax_kms=0.09,zmin=2.19,zmax=None):
+        """Read measured P1D from file.
+            - diag_cov: for now, use diagonal covariance
+            - kmax_kms: limit to low-k where we trust emulator """
+
+
+        # read redshifts, wavenumbers, power spectra and covariance matrices
+        z,k,Pk,cov=read_from_file(diag_cov,kmax_kms)
+
+        # drop low-z or high-z bins
+        if zmin or zmax:
+            z,k,Pk,cov=_drop_zbins(z,k,Pk,cov,zmin,zmax)
+
+        super().__init__(z,k,Pk,cov)
+
+        return
+
+
+def read_from_file(diag_cov, kmax_kms):
+    """Read file containing mock P1D"""
+
+    # folder storing P1D measurement
+    datadir=BaseDataP1D.BASEDIR + '/Karacayli2023/'
+    fname = datadir + "/desi-edrp-lyasb1subt-p1d-detailed-results.txt"
+
+    with open(fname) as _:
+        names = _.readline()[1:].strip().split()
+
+    # start by reading the file with measured band power
+    # z k1 k2 kc Pfid ThetaP p_final e_stat p_raw p_noise p_fid_qmle p_smooth e_n_syst e_res_syst e_cont_syst e_dla_syst p_sb1 e_sb1_stat e_total
+    data = pandas.read_table(
+        fname, comment='#', names=names, delim_whitespace=True
+    ).to_records(index=False)
+    
+    if diag_cov:
+        cov_full = np.diag(data['e_total']**2)
+    else:
+        fname_cov = datadir + "/desi-edrp-lyasb1subt-cov-total-offdiag-results.txt"
+        cov_full = np.loadtxt(fname_cov)
+
+    zbins = np.unique(data['z'])
+    kbins = np.unique(data['kc'])
+    Nk = kbins.size
+    Nz = zbins.size
+
+    print('Nz = {} , Nk = {}'.format(Nz, Nk))
+    Pk = []
+    cov = []
+    for iz in range(Nz):
+        w = np.isclose(data['z'], zbins[iz])
+        tmp_d = np.zeros(Nk)
+        _nk = w.sum()
+        tmp_d[:_nk] = data['p_final'][w]
+        Pk.append(tmp_d)
+        
+        tmp_cov = np.zeros((Nk, Nk))
+        # Fill non-existing k bins with inf
+        np.fill_diagonal(tmp_cov, 1e12)
```

Comment:
What's the motivation for filling these bins with 1e12? I think it would be better to leave these with nans or zeros, otherwise giant error bars appear in our plots.

Reply:
I would imagine that the motivation is that you can still invert it. 

We could mask these data points at the plotting stage, but the fits should be unaffected by this. 

If we added zeros or NaNs, this choice would impact the fits.

One way or the other, we need to make sure that anytime P1D is used one checks for NaNs or large values and knows how to mask properly...

Reply:
Ok, let's leave it like this then. I would make this very clear in the file though, it took me a while to understand what was happening.

Reply:
Andreu is correct. Added a comment. Note I now changed them to np.inf